state/factory/workingset.go

```diff
@@ -92,25 +92,20 @@ func NewWorkingSet(
 // LoadOrCreateAccountState loads existing or adds a new account state with initial balance to the factory
 // addr should be a bech32 properly-encoded string
 func (ws *workingSet) LoadOrCreateAccountState(addr string, init *big.Int) (*state.Account, error) {
-	addrHash, err := iotxaddress.AddressToPKHash(addr)
+	account, err := ws.CachedAccountState(addr)
 	if err != nil {
-		return nil, err
-	}
-	s, err := ws.CachedState(addrHash, &state.Account{})
-	switch {
-	case errors.Cause(err) == state.ErrStateNotExist:
-		account := state.Account{
-			Balance:      big.NewInt(0).SetBytes(init.Bytes()),
+		return nil, errors.Wrapf(err, "failed to get account of %s from cached account", addr)
+	} else if account == state.EmptyAccount {
+		account = &state.Account{
+			Balance:      init,
 			VotingWeight: big.NewInt(0),
 		}
-		ws.cachedStates[addrHash] = &account
-		return &account, nil
-	case err != nil:
-		return nil, errors.Wrapf(err, "failed to get account of %x from cached account", addrHash)
-	}
-	account, err := stateToAccountState(s)
-	if err != nil {
-		return nil, err
+		addrHash, err := iotxaddress.AddressToPKHash(addr)
+		if err != nil {
```

Comment:
no need to check error here
it is already checked in CachedAccountState() above